### ISSUE TITLE
test: test_node_base_split_region_right_derive failed

### ISSUE THREAD

**Issue by AndreMouche** [pingcap/tikv#2487]
Hi, @overvenus 
  `test_node_base_split_region_right_derive` failed sometime in my environment, here is the details:
```
thread 'raftstore-1::test_node_base_split_region_right_derive' panicked at 'failed to split: message: "[region 1] 1 epoch changed conf_ver: 1 version: 3 != conf_ver: 1 version: 2, need re-check later"', tests/raftstore/cluster.rs:769:24
test raftstore::test_split_region::test_node_quick_election_after_split ... ok
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
stack backtrace:
   0: std::sys::imp::backtrace::tracing::imp::unwind_backtrace
             at /checkout/src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::_print
             at /checkout/src/libstd/sys_common/backtrace.rs:71
   2: std::panicking::default_hook::{{closure}}
             at /checkout/src/libstd/sys_common/backtrace.rs:60
             at /checkout/src/libstd/panicking.rs:380
   3: std::panicking::default_hook
             at /checkout/src/libstd/panicking.rs:396
   4: tikv::util::panic_hook::track_hook::{{closure}}
             at src/util/panic_hook.rs:61
   5: <std::thread::local::LocalKey<T>>::try_with
             at /checkout/src/libstd/thread/local.rs:362
   6: <std::thread::local::LocalKey<T>>::with
             at /checkout/src/libstd/thread/local.rs:276
   7: tikv::util::panic_hook::track_hook
             at src/util/panic_hook.rs:55
   8: core::ops::function::Fn::call
             at /checkout/src/libcore/ops/function.rs:48
   9: std::panicking::rust_panic_with_hook
             at /checkout/src/libstd/panicking.rs:611
  10: std::panicking::begin_panic
             at /checkout/src/libstd/panicking.rs:571
  11: std::panicking::begin_panic_fmt
             at /checkout/src/libstd/panicking.rs:521
  12: <tests::raftstore::cluster::Cluster<T>>::must_split::{{closure}}
             at tests/raftstore/cluster.rs:769
  13: core::ops::function::FnOnce::call_once
             at /checkout/src/libcore/ops/function.rs:143
  14: <F as alloc::boxed::FnBox<A>>::call_box
             at /checkout/src/liballoc/boxed.rs:682
  15: <alloc::boxed::Box<alloc::boxed::FnBox<A, Output$u3d$R$GT$$u20$$u2b$$u20$core..marker..Send$u20$$u2b$$u20$$u27$a$GT$$u20$as$u20$core..ops..function..FnOnce$LT$A$GT$$GT$::call_once
             at /checkout/src/liballoc/boxed.rs:702
  16: <tikv::raftstore::store::store::Store<T, C>>::on_prepare_split_region::{{closure}}
             at ./src/raftstore/store/store.rs:1844
  17: <core::option::Option<T>>::map
             at /checkout/src/libcore/option.rs:398
  18: <tikv::raftstore::store::store::Store<T, C>>::on_prepare_split_region
             at ./src/raftstore/store/store.rs:1844
  19: <tikv::raftstore::store::store::Store<T, C> as mio::handler::Handler>::notify
             at ./src/raftstore/store/store.rs:2470
  20: <mio::event_loop::EventLoop<H>>::notify
             at /home/jenkins/.cargo/registry/src/github.com-1ecc6299db9ec823/mio-0.5.1/src/event_loop.rs:350
  21: <mio::event_loop::EventLoop<H>>::run_once
             at /home/jenkins/.cargo/registry/src/github.com-1ecc6299db9ec823/mio-0.5.1/src/event_loop.rs:299
  22: <mio::event_loop::EventLoop<H>>::run
             at /home/jenkins/.cargo/registry/src/github.com-1ecc6299db9ec823/mio-0.5.1/src/event_loop.rs:247
  23: <tikv::raftstore::store::store::Store<T, C>>::run
             at ./src/raftstore/store/store.rs:538
  24: <tikv::server::node::Node<C>>::start_store::{{closure}}
             at ./src/server/node.rs:356
thread 'raftstore::test_split_region::test_node_base_split_region_right_derive' panicked at 'called `Result::unwrap()` on an `Err` value: Timeout("request timeout for Duration { secs: 5, nanos: 0 }")', /checkout/src/libcore/result.rs:860:4
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
stack backtrace:
   0: std::sys::imp::backtrace::tracing::imp::unwind_backtrace
             at /checkout/src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::_print
             at /checkout/src/libstd/sys_common/backtrace.rs:71
   2: std::panicking::default_hook::{{closure}}
             at /checkout/src/libstd/sys_common/backtrace.rs:60
             at /checkout/src/libstd/panicking.rs:380
   3: std::panicking::default_hook
             at /checkout/src/libstd/panicking.rs:396
   4: tikv::util::panic_hook::track_hook::{{closure}}
             at src/util/panic_hook.rs:61
   5: <std::thread::local::LocalKey<T>>::try_with
             at /checkout/src/libstd/thread/local.rs:362
   6: <std::thread::local::LocalKey<T>>::with
             at /checkout/src/libstd/thread/local.rs:276
   7: tikv::util::panic_hook::track_hook
             at src/util/panic_hook.rs:55
   8: core::ops::function::Fn::call
             at /checkout/src/libcore/ops/function.rs:48
   9: std::panicking::rust_panic_with_hook
             at /checkout/src/libstd/panicking.rs:611
  10: std::panicking::begin_panic
             at /checkout/src/libstd/panicking.rs:571
  11: std::panicking::begin_panic_fmt
             at /checkout/src/libstd/panicking.rs:521
  12: rust_begin_unwind
             at /checkout/src/libstd/panicking.rs:497
  13: core::panicking::panic_fmt
             at /checkout/src/libcore/panicking.rs:71
  14: core::result::unwrap_failed
             at /checkout/src/libcore/macros.rs:31
  15: <core::result::Result<T, E>>::unwrap
             at /checkout/src/libcore/result.rs:738
  16: <tests::raftstore::cluster::Cluster<T>>::query_leader
             at tests/raftstore/cluster.rs:281
  17: <tests::raftstore::cluster::Cluster<T>>::leader_of_region
             at tests/raftstore/cluster.rs:319
  18: <tests::raftstore::cluster::Cluster<T>>::call_command_on_leader
             at tests/raftstore/cluster.rs:233
  19: <tests::raftstore::cluster::Cluster<T>>::request
             at tests/raftstore/cluster.rs:516
  20: <tests::raftstore::cluster::Cluster<T>>::must_put_cf
             at tests/raftstore/cluster.rs:592
  21: <tests::raftstore::cluster::Cluster<T>>::must_put
             at tests/raftstore/cluster.rs:588
  22: tests::raftstore::test_split_region::test_base_split_region
             at tests/raftstore/test_split_region.rs:80
  23: tests::raftstore::test_split_region::test_node_base_split_region_right_derive
             at tests/raftstore/test_split_region.rs:118
  24: <F as test::FnBox<T>>::call_box
             at /checkout/src/libtest/lib.rs:1477
             at /checkout/src/libcore/ops/function.rs:143
             at /checkout/src/libtest/lib.rs:138
  25: __rust_maybe_catch_panic
             at /checkout/src/libpanic_unwind/lib.rs:98
thread 'raftstore::test_split_region::test_node_base_split_region_right_derive' panicked at 'called `Result::unwrap()` on an `Err` value: Other(Closed)', /checkout/src/libcore/result.rs:860:4
stack backtrace:
   0:     0x7f9908260aa3 - std::sys::imp::backtrace::tracing::imp::unwind_backtrace::hc32b3550469bd7ec
                               at /checkout/src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1:     0x7f990825cda4 - std::sys_common::backtrace::_print::h18ec9a3165b0d7ba
                               at /checkout/src/libstd/sys_common/backtrace.rs:71
   2:     0x7f9908264243 - std::panicking::default_hook::{{closure}}::hcead061326bfc66a
                               at /checkout/src/libstd/sys_common/backtrace.rs:60
                               at /checkout/src/libstd/panicking.rs:380
   3:     0x7f9908263fa2 - std::panicking::default_hook::h6a8c02fe69584b6b
                               at /checkout/src/libstd/panicking.rs:396
   4:     0x7f9906c3e3d9 - tikv::util::panic_hook::track_hook::{{closure}}::hc9b8e85b5edbe162
                               at src/util/panic_hook.rs:61
   5:     0x7f99069ed75d - <std::thread::local::LocalKey<T>>::try_with::ha104633481d3e320
                               at /checkout/src/libstd/thread/local.rs:362
   6:     0x7f99069ed21c - <std::thread::local::LocalKey<T>>::with::h2f91a4b682816b30
                               at /checkout/src/libstd/thread/local.rs:276
   7:     0x7f9906c3e2fe - tikv::util::panic_hook::track_hook::h5bce1462152058aa
                               at src/util/panic_hook.rs:55
   8:     0x7f9906a6be48 - core::ops::function::Fn::call::h523d7c556096ceb5
                               at /checkout/src/libcore/ops/function.rs:48
   9:     0x7f9908264786 - std::panicking::rust_panic_with_hook::h536d0075488cfad3
                               at /checkout/src/libstd/panicking.rs:611
  10:     0x7f9908264614 - std::panicking::begin_panic::h59163d4c7e23a4b5
                               at /checkout/src/libstd/panicking.rs:571
  11:     0x7f9908264519 - std::panicking::begin_panic_fmt::h226cbf2ee66e31a5
                               at /checkout/src/libstd/panicking.rs:521
  12:     0x7f99082644aa - rust_begin_unwind
                               at /checkout/src/libstd/panicking.rs:497
  13:     0x7f9908292ba0 - core::panicking::panic_fmt::h232071f163afd174
                               at /checkout/src/libcore/panicking.rs:71
  14:     0x7f990615c3c4 - core::result::unwrap_failed::h3969aaa97db20678
                               at /checkout/src/libcore/macros.rs:31
  15:     0x7f9905ebdc36 - <core::result::Result<T, E>>::unwrap::h197d07fcd227ff0c
                               at /checkout/src/libcore/result.rs:738
  16:     0x7f99067762b8 - <tests::raftstore::node::NodeCluster as tests::raftstore::cluster::Simulator>::stop_node::h7c66e3df0684a4b9
                               at tests/raftstore/node.rs:233
  17:     0x7f990675512a - <tests::raftstore::cluster::Cluster<T>>::stop_node::hc416382ac2150f08
                               at tests/raftstore/cluster.rs:177
  18:     0x7f9906760b4c - <tests::raftstore::cluster::Cluster<T>>::shutdown::hb5c291789d92eebb
                               at tests/raftstore/cluster.rs:467
  19:     0x7f9906772f7c - <tests::raftstore::cluster::Cluster<T> as core::ops::drop::Drop>::drop::hec46ef738bb6d90b
                               at tests/raftstore/cluster.rs:857
  20:     0x7f99060c36e4 - core::ptr::drop_in_place::h66117c2207abaef4
                               at /checkout/src/libcore/ptr.rs:61
  21:     0x7f99067c466b - tests::raftstore::test_split_region::test_node_base_split_region_right_derive::hcac07bc0369ec344
                               at tests/raftstore/test_split_region.rs:119
  22:     0x7f99069026b1 - <F as test::FnBox<T>>::call_box::h5e82055b2788b0b0
                               at /checkout/src/libtest/lib.rs:1477
                               at /checkout/src/libcore/ops/function.rs:143
                               at /checkout/src/libtest/lib.rs:138
  23:     0x7f990826b89c - __rust_maybe_catch_panic
                               at /checkout/src/libpanic_unwind/lib.rs:98
  24:     0x7f99068f302c - std::sys_common::backtrace::__rust_begin_short_backtrace::hed881682075c7641
                               at /checkout/src/libstd/panicking.rs:458
                               at /checkout/src/libstd/panic.rs:361
                               at /checkout/src/libtest/lib.rs:1416
                               at /checkout/src/libstd/sys_common/backtrace.rs:136
  25:     0x7f99068f3df2 - std::panicking::try::do_call::hfb55048cb24e8c46
                               at /checkout/src/libstd/thread/mod.rs:364
                               at /checkout/src/libstd/panic.rs:296
                               at /checkout/src/libstd/panicking.rs:479
  26:     0x7f990826b89c - __rust_maybe_catch_panic
                               at /checkout/src/libpanic_unwind/lib.rs:98
  27:     0x7f99068fbff2 - <F as alloc::boxed::FnBox<A>>::call_box::h38e755be7f3e288a
                               at /checkout/src/libstd/panicking.rs:458
                               at /checkout/src/libstd/panic.rs:361
                               at /checkout/src/libstd/thread/mod.rs:363
                               at /checkout/src/liballoc/boxed.rs:682
  28:     0x7f99082633ab - std::sys::imp::thread::Thread::new::thread_start::h0aeebbbbd5e5d9cd
                               at /checkout/src/liballoc/boxed.rs:692
                               at /checkout/src/libstd/sys_common/thread.rs:21
                               at /checkout/src/libstd/sys/unix/thread.rs:84
  29:     0x7f9904727dc4 - start_thread
  30:     0x7f990424076c - clone
  31:                0x0 - <unknown>
thread panicked while panicking. aborting.
error: An unknown error occurred

```

**Comment by AndreMouche** [comment]
related test http://45.249.247.133:8080/blue/organizations/jenkins/tikv_ghpr_unit_test/detail/tikv_ghpr_unit_test/120/pipeline

**Comment by overvenus** [comment]
I think it has been fixed by #2595.
